runtime/wasm/src/mapping.rs

```diff
@@ -61,25 +61,15 @@ pub fn spawn_module<C: Blockchain>(
                     trigger,
                     result_sender,
                 } = request;
-                let logger = ctx.logger.cheap_clone();
 
-                // Start the WASM module runtime.
-                let section = host_metrics.stopwatch.start_section("module_init");
-                let module = WasmInstance::from_valid_module_with_ctx(
+                let result = instantiate_module_and_handle_trigger(
                     valid_module.cheap_clone(),
                     ctx,
+                    trigger,
                     host_metrics.cheap_clone(),
                     timeout,
                     experimental_features,
-                )?;
```

Comment:
This early return was the issue, I fixed it by refactoring this into a separate fn.